### PR TITLE
 Adjust create new entity modal 

### PR DESCRIPTION
**Pull request by Badlapje**
Prior to this change, the create new entity modal only inquired about the desired protocol.

This change adds an inquirey for the environment and the template to copy from.

[See the story in pivotal for more info](https://www.pivotaltracker.com/story/show/165599582)

### CODE REVIEW

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityActions/actionsForDetail.html.twig [hunk 1]
```diff
@@ -2,6 +2,6 @@
     {% include "@Dashboard/EntityActions/overviewAction.html.twig" %}
     {% include "@Dashboard/EntityActions/editAction.html.twig" with {action: entity} %}
     {% include "@Dashboard/EntityActions/secretResetAction.html.twig" with {action: entity} %}
-    {% include "@Dashboard/EntityActions/aclAction.html.twig" with {action: entity} %}
+    {% include "@Dashboard/EntityActions/aclAction.html.twig" with {action: en tity} %}
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityActions/actionsForDetail.html.twig:5]: Again with the profanity!

**Review by MKodde — CHANGES_REQUESTED** [review]
This was just a code review. Thinks look great, I found some issues you might want to address.

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityActions/actionsForDetail.html.twig (hunk 1 above)
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityActions/actionsForDetail.html.twig]: 😂 

Although in my defence I should point out that it's not profane to talk about a tity (as in a small tithe).  That'd be tittie you're thinking about 😇 

Fixed it!

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityModal/addEntityModal.html.twig [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{% block body %}
+    <div class="row entity-type">
+            <fieldset class="add-entity-fieldset add-entity-environment hidden" aria-describedby="fieldset1">
+                <h5 class="add-entity-question" id="fieldset1">Environment?</h5>
+                <ul class="add-entity-fieldset-fields">
+                    {% set fieldName = guid ~ '_environment' %}
+                    <li class="add-entity-field">
+                        {% set fieldId = guid ~ '_prod' %}
+                        <input class="add-entity-radio" type="radio" name={{ fieldName }} id="{{ fieldId }}" value="production"{% if not productionEnabled %} disabled="disabled"{% endif %}{% if environment is same as('production') and productionEnabled %} checked="checked"{% endif %}>
+                        <label class="add-entity-label" for="{{ fieldId }}">production</label>
```
[the remark this answers is not in the document]
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityModal/addEntityModal.html.twig:10]: edited

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityModal/addEntityModal.html.twig [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                        {% set fieldId = guid ~ '_' ~ value %}
+                        <li class="add-entity-field">
+                            <input class="add-entity-radio" type="radio" name={{ fieldName }} id={{ fieldId }} value={{ value }}{% if loop.first %} checked="checked"{% endif %}>
+                            <label class="add-entity-label" for={{ fieldId }}>{{ title|trans }}</label>
+                        </li>
+                    {% endfor %}
+                </ul>
+            </fieldset>
+            <fieldset class="add-entity-fieldset add-entity-template" aria-describedby="fieldset3">
+                <h5 class="add-entity-question" id="fieldset3">From template?</h5>
```
[the remark this answers is not in the document]
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityModal/addEntityModal.html.twig:33]: i think i got them all :-)

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php [hunk 4]
```diff
@@ -90,32 +91,59 @@ public function __construct(
...
-        if ($request->request->has('dashboard_bundle_choose_entity_type_1')) {
-            $form = $this->createForm(ChooseEntityTypeType::class, $command);
-        }
+            if ($envFirst === $envSecond) {
+                return ($first->getName() < $second->getName()) ? -1 : 1;
+            }
+
+            return ($envFirst < $envSecond) ? -1 : 1;
+        };
+        usort($entityList, $sortEntitiesByEnvironment);
```
[the remark this answers is not in the document]
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php:116]: moved it to the EntityList class, where i reckon it fits in nicely.

File: src/Surfnet/ServiceProviderDashboard/Application/ViewObject/EntityList.php [hunk 5]
```diff
@@ -36,15 +36,41 @@ public function __construct(array $entities)
...
+     * @param String $sortBy
      * @return Entity[]
      */
-    public function getEntities()
+    public function getEntities(String $sortBy = '')
     {
+        if (strlen($sortBy) > 0) {
+            $entities = $this->entities;
+            switch ($sortBy) {
+                case 'environment':
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Application/ViewObject/EntityList.php:47]: Please add a class constant for this value. Also, I'm not stoked on adding the sorting logic to a getter. A getter should be dumb as possible. Maybe decouple this from the getter and add a (chainable) sort method? That way, the other getters could also benefit from the sorting logic. And the getter has a single responsibility.

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityDetail/detail.html.twig [hunk 6]
```diff
@@ -5,6 +5,7 @@
 
     <h1>{% block page_heading %}{{ 'entity.detail.title'|trans }}{% endblock %}</h1>
 
+    <h2>{{ isAdmin }}</h2>
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityDetail/detail.html.twig:8]: I think this was carried over from an old branch? This `isAdmin` h2 was removed in another dev-effort IIRC

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/translations/messages.en.yml [hunk 7]
```diff
@@ -581,3 +581,12 @@ entity.edit.form.attributes.saml20.eduPersonTargetedIDAttribute: Edu person targ
...
 entity.actions.modal.oidc_client_reset.info.html: <p>Are you sure you would like to reset the openid connect client secret.</p>
+entity.add.environment.title: Environment?
+entity.add.environment.production: production
+entity.add.environment.test: test
+entity.add.protocol.title: Protocol?
+entity.add.template.title: From template?
+entity.add.template.no: No, create blank registration form.
+entity.add.template.yes: Yes
+entity.add.create: Create
+entity.add.cancel: Cancel
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/translations/messages.en.yml:592]: :100:

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php [hunk 8]
```diff
@@ -90,32 +91,47 @@ public function __construct(
      * @param string $inputId
      * @return array|RedirectResponse
      */
-    public function typeAction(Request $request, $serviceId, $targetEnvironment, string $inputId)
+    public function typeAction(Request $request, $serviceId, string $targetEnvironment, string $inputId)
     {
         $service = $this->authorizationService->changeActiveService($serviceId);
 
         $choices = $this->protocolChoiceFactory->buildOptions();
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php:98]: I guess the choices, and the command declared below are now unused code. Perhaps the command can be rid of entirely? 

:wastebasket: time!

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php [hunk 9]
```diff
@@ -124,6 +140,10 @@ public function typeAction(Request $request, $serviceId, $targetEnvironment, str
             'serviceId' => $service->getId(),
             'environment' => $targetEnvironment,
             'inputId' => $inputId,
+            'protocols' => $command->getProtocolChoices(),
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php:143]: No need to use the command for this

**Review by MKodde — CHANGES_REQUESTED** [review]
Using my morning freshness, I found some more things that can be improved in this PR.

I like the choice you made to move the sorting logic to the list. Maybe create a sort method on the list instead of implementing it into the getter. I also found some more cleanup opportunities.

File: src/Surfnet/ServiceProviderDashboard/Application/ViewObject/EntityList.php (hunk 5 above)
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Application/ViewObject/EntityList.php:47]: i like the idea of a chainable sort method, brilliant 💯

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityDetail/detail.html.twig (hunk 6 above)
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Resources/views/EntityDetail/detail.html.twig:8]: /me glares at git

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php (hunk 9 above)
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php:143]: nice spot!

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php (hunk 8 above)
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php:98]: 't has been incinerated!

Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php:98]: though the choices are still needed for construction of the form.

**Review by MKodde — APPROVED** [review]
Nailed it!

**Pull request merged**